tooling/bundler/src/bundle.rs

```diff
@@ -93,9 +97,23 @@ pub fn bundle_project(settings: Settings) -> crate::Result<Vec<Bundle>> {
       PackageType::AppImage => linux::appimage::bundle_project(&settings)?,
 
       // updater is dependant of multiple bundle, we send our bundles to prevent rebuilding
-      PackageType::Updater => updater_bundle::bundle_project(&settings, &bundles)?,
+      PackageType::Updater => {
+        if !package_types.iter().any(|p| {
+          matches!(
+            p,
+            PackageType::AppImage
+              | PackageType::MacOsBundle
+              | PackageType::Nsis
+              | PackageType::WindowsMsi
+          )
+        }) {
+          warn!("The updater bundle target exists but couldn't find any updater-enabled target, so the updater artifacts won't be generated. Please add one of these targets as well: app, appimage, msi, nsis");
+          continue;
+        }
+        updater_bundle::bundle_project(&settings, &bundles)?
```

Comment:
I have just found out that `package_types` is never updated when a macOS DMG bundle is generated (see lines 77-85). Therefore, the updater bundle will not be generated if the macOS .app bundle is not explicitly set in `package_types`, even though the DMG bundle implies the generation of an app bundle and thus `package_types` does not need to explicitly contain a macOS .app bundle.

I'd suggest adding a `| PackageType::Dmg` branch to the `matches!` macro here to better handle this edge case.